sdk-setup/src/mb2

```diff
@@ -1856,104 +1856,107 @@ run_build_requires__process_args() {
-            fi
+        if ! gpg2 --quiet --import "${shared_key}"; then 
+            fatal "Cannot sign packages: Internal error: Failed to import GPG key from file '$shared_key'."
         fi
+    fi
 
-        if ! gpg2 --list-secret-keys "$OPT_PACKAGE_SIGN_USER" &>/dev/null; then
-            notice "Cannot sign packages: No GPG key found for the user '$OPT_PACKAGE_SIGN_USER'."
-            return 1
-        fi
+    if ! gpg2 --list-secret-keys "$OPT_PACKAGE_SIGN_USER" &>/dev/null; then
+        fatal "Cannot sign packages: No GPG key found for the user '$OPT_PACKAGE_SIGN_USER'."
+    fi
 
-        extra_args+=(--define "_gpg_name $OPT_PACKAGE_SIGN_USER")
+    SIGN_EXTRA_ARGS+=(--define "_gpg_name $OPT_PACKAGE_SIGN_USER")
 
-        local passphrase=
+    if [[ $OPT_PACKAGE_SIGN_PASSPHRASE_FILE ]]; then
+        SIGN_PASSPHRASE=$(<"$OPT_PACKAGE_SIGN_PASSPHRASE_FILE")
+    else
+        SIGN_PASSPHRASE=$OPT_PACKAGE_SIGN_PASSPHRASE
+    fi
 
-        if [[ $OPT_PACKAGE_SIGN_PASSPHRASE_FILE ]]; then
-            passphrase=$(<"$OPT_PACKAGE_SIGN_PASSPHRASE_FILE")
+    if ! gpg2 --batch --status-fd 1 --sign --local-user "$OPT_PACKAGE_SIGN_USER" \
+            --output - --passphrase-fd 3 3<<<"$SIGN_PASSPHRASE" </dev/null &>/dev/null; then
+        if [[ ! $SIGN_PASSPHRASE ]]; then
+            fatal "Cannot sign packages: The selected GPG key is passphrase protected" \
+                "and no passphrase was specified."
         else
-            passphrase=$OPT_PACKAGE_SIGN_PASSPHRASE
+            fatal "Cannot sign packages: The given passphrase is not a correct passphrase" \
+                "for the '$OPT_PACKAGE_SIGN_USER' key."
         fi
+    fi
+}
 
-        if ! gpg2 --batch --status-fd 1 --sign --local-user "$OPT_PACKAGE_SIGN_USER" \
-                --output - --passphrase-fd 3 3<<<"$passphrase" </dev/null &>/dev/null; then
-            if [[ ! $passphrase ]]; then
-                notice "Cannot sign packages: The selected GPG key is passphrase protected" \
-                    "and no passphrase was specified."
-            else
-                notice "Cannot sign packages: The given passphrase is not a correct passphrase" \
-                    "for the '$OPT_PACKAGE_SIGN_USER' key."
-            fi
-            return 1
-        fi
+sign_rpms_native() {
+    local rpms=("$@")
 
-        if [[ $passphrase ]]; then
-            tmpdir=$(mktemp -d) || return
-            chmod 0700 "$tmpdir" || return
-            local passphrase_file=$tmpdir/passphrase
-            cat >"$passphrase_file" <<<"$passphrase" || return
-            extra_args+=(--define "_gpg_sign_cmd_extra_args --batch --passphrase-file '$passphrase_file'")
+    local tmpdir=
+    sign_rpms_native_cleanup() (
+        trap 'echo cleaning up...' INT TERM HUP
+        if [[ $tmpdir ]]; then
+            rm -rf "$tmpdir"
         fi
+    )
+    trap 'sign_rpms_native_cleanup; trap - RETURN' RETURN
+    trap 'return 1' INT TERM HUP
+
+    if [[ $SIGN_PASSPHRASE ]]; then
```

Comment:
I would feel a tiny bit more safer if the passphrase read from file was discarded immediately after it's use, so I would simply repeat this piece of code here instead of passing it by SIGN_PASSPHRASE:

```
    local passphrase=
    if [[ $OPT_PACKAGE_SIGN_PASSPHRASE_FILE ]]; then
            passphrase=$(<"$OPT_PACKAGE_SIGN_PASSPHRASE_FILE")
    else
            passphrase=$OPT_PACKAGE_SIGN_PASSPHRASE
    fi
```